packages/gatsby/src/schema/__tests__/node-model.js

```diff
@@ -884,6 +923,44 @@ describe(`NodeModel`, () => {
       expect(result2.map(node => node.id)).toEqual([`id4`, `id5`])
     })
 
+    it(`sorts correctly by fields with custom resolvers`, async () => {
+      // See https://github.com/gatsbyjs/gatsby/issues/28047
+      nodeModel.replaceFiltersCache()
+
+      // This is required to setup a state after which the error reveals itself
+      const result1 = await nodeModel.runQuery(
+        {
+          query: {
+            filter: { id: { regex: `/non-existing/` } },
+          },
+          firstOnly: true,
```

Comment:
Isn't `firstOnly` the bigger problem here? Does the problem still happen when it's not `firstOnly`? It's been a few months since I looked at this but I thought `firstOnly` would shortcut some steps. Maybe another test case, where `firstOnly` is `false` for both cases (and maybe one where it is `true` in both cases) is useful here.

Reply:
I've checked with `firstOnly: false` and there is no difference. I picked `firstOnly` just to mimic the reproduction from the original issue.

The critical thing required to reproduce this is to populate `filtersCache` and keep `__gatsby_resolved` equal to `undefined` at the same time (so first query must have no filters/sorts by fields with custom resolvers)